### PR TITLE
Added support for custom kafka headers

### PR DESCRIPTION
**Pull request by bbalser**

### CODE REVIEW

File: lib/elsa/producer.ex [hunk 1]
```diff
@@ -73,10 +74,8 @@ defmodule Elsa.Producer do
 
   defp produce_sync_while_successful(registry, topic, message_chunks) do
     Enum.reduce_while(message_chunks, {:ok, 0}, fn {partition, chunk}, {:ok, messages_sent} ->
-      total_size = Enum.reduce(chunk, 0, fn {key, value}, acc -> acc + byte_size(key) + byte_size(value) end)
```
Comment (jeffgrunewald) [inline lib/elsa/producer.ex]: I assume this was just taking extra time.

File: lib/elsa/util.ex [hunk 2]
```diff
@@ -163,6 +163,15 @@ defmodule Elsa.Util do
...
+  defp get_byte_size(%{key: key, value: value} = msg) do
+    header_size =
+      Map.get(msg, :headers, [])
+      |> Enum.map(fn {key, value} -> byte_size(key) + byte_size(value) end)
+      |> Enum.sum()
+
+    byte_size(key) + byte_size(value) + header_size
+  end
+
   defp get_byte_size({one, two}) do
```
Comment (jeffgrunewald) [inline lib/elsa/util.ex]: Is this function head for `get_byte_size` still valid? Looks like you’ve converted all the messages from a tuple to a map now.

File: lib/elsa/producer.ex (hunk 1 above)
Comment (bbalser) [reply lib/elsa/producer.ex]: yeah, it was just for a log message, which didn't seem worth it.

File: lib/elsa/util.ex (hunk 2 above)
Comment (bbalser) [reply lib/elsa/util.ex]: good catch, I will drop the others

**Pull request merged**